roles/translator/src/upstream_sv2/upstream.rs

```diff
@@ -141,12 +142,24 @@ impl Upstream {
         extranonce_sender: Sender<ExtendedExtranonce>,
         target: Arc<Mutex<Vec<u8>>>,
     ) -> ProxyResult<Arc<Mutex<Self>>> {
-        // Connect to the SV2 Upstream role
-        let socket = TcpStream::connect(address).await?;
+        // Connect to the SV2 Upstream role retry connection every 5 seconds.
+        let socket = loop {
+            match TcpStream::connect(address).await {
+                Ok(socket) => break socket,
+                Err(e) => {
+                    error!(
+                        "Failed to connect to Upstream role at {}, retrying in 5s: {}",
+                        address, e
+                    );
+
+                    sleep(Duration::from_secs(5));
```

Comment:
maybe we want to parametrize the sleep duration?

Reply:
I'm going to skip this for now. This is going to be a pretty corner case where the pool isn't already running, the only thing I could think of doing more is to make the retries back off - so it starts at 1s and then fades back with every try - both really feel like overkill for this corner case.